firestore/add_data.rb

```diff
@@ -0,0 +1,190 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/cloud/firestore"
+require "date"
+
+def set_document project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  # [START fs_set_document]
+  city_ref = firestore.doc "cities/LA"
+
+  data = {
+    name: "Los Angeles",
+    state: "CA",
+    country: "USA"
+  }
+
+  city_ref.set(data)
+  # [END fs_set_document]
+  puts "Set data for the LA document in the cities collection."
+end
+
+def update_create_if_missing project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  # [START fs_update_create_if_missing]
+  city_ref = firestore.doc "cities/LA"
+  city_ref.set({ capital: false }, merge: true)
+  # [END fs_update_create_if_mising]
+  puts "Merged data into the LA document in the cities collection."
+end
+
+def set_document_data_types project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  # [START fs_set_document_data_types]
+  doc_ref = firestore.doc "data/one"
+
+  data = {
+    stringExample: "Hello, World!",
+    booleanExample: true,
+    numberExample: 3.14159265,
+    dateExample: DateTime.now,
+    arrayExample: [5, true, "hello"],
+    nullExample: nil,
+    objectExample: {
+      a: 5,
+      b: true
+    }
+  }
+
+  doc_ref.set(data)
+  # [END fs_set_document_data_types]
+  puts "Set multiple data-type data for the one document in the data collection."
+end
+
+def set_requires_id project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  data = {
+    name: "Phuket",
+    country: "Thailand"
+  }
+  # [START fs_set_requires_id]
+  city_ref = firestore.doc "cities/new-city-id"
+  city_ref.set(data)
+  # [END fs_set_requires_id]
+  puts "Added document with ID: new-city-id."
+end
+
+def add_doc_data_with_auto_id project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  # [START fs_add_doc_data_with_auto_id]
+  data = {
+    name: "Tokyo",
+    country: "Japan"
+  }
+
+  cities_ref = firestore.col "cities"
+
+  added_doc_ref = cities_ref.add(data)
+  puts "Added document with ID: #{added_doc_ref.document_id}."
+  # [END fs_add_doc_data_with_auto_id]
+end
+
+def add_doc_data_after_auto_id project_id:
+  # project_id = "Your Google Cloud Project ID"
+  firestore = Google::Cloud::Firestore.new(project_id: project_id)
+  data = {
```

Comment:
consider making creation of data variable consistent across add_doc_data_after_auto_id and update_doc